CMSgov/easi-app
# Customize notification emails' subject/text

# EASI-2235

## Changes and Description

- Change the subjects of notification emails to more generic text, to account for them being sent to people besides an intake's requester.
- Change notification emails to include text explaining why the recipient is receiving the email.
- Change notification emails to include text explaining how the original requester can view the request, with a link.

## Notes

* I tried to use a consistent order for the parameters of the email methods - context, recipient(s), intake ID, project name, requester, than all other parameters.

## How to test this change

* Requirements when running locally
    * Connect to the VPN (in order to be able to connect to CEDAR LDAP). 
    * In `src/views/FlagsWrapper/index.tsx`, change the default value for `notifyMultipleRecipients` from `false` to `true`.
    * In `pkg/server/routes.go`, comment out the block that sets `cedarLDAPClient = local.NewCedarLdapClient(s.logger)`.
    * In `pkg/services/config.go`, change `checkBoolFeatureFlag()` to unconditionally return true.
* When creating intakes to test with, make sure to use local auth with a valid EUA ID, so there's an associated email address.
* To verify email text, check Mailcatcher. To check all emails covered by this PR:
    * issue_lcid - Create an intake, go to the admin view, go to Actions, issue a lifecycle ID.
    * extend_lcid - Issue a lifecycle ID, then go to Actions again, and extend the lifecycle ID.
    * intake_review/system_intake_review - Create an intake, go to the admin view, go to Actions, and select "Provide GRT feedback and keep working on draft business case".
    * reject_request
        * Create an intake.
        * Go to the admin view, go to Actions, select "Request a draft business case", and complete form.
        * Go to the IT Governance tab, select the intake to access its task list, click "Start" in the section for "Prepare your Business Case for the GRT".
        * Fill out the "Requester Phone Number" field, then click "Save & Exit".
        * In the admin view, go to Actions, and select "Project not approved by GRB".
